src/components/transport_manager/src/usb/libusb/usb_connection.cc

```diff
@@ -241,22 +250,37 @@ TransportAdapter::Error UsbConnection::SendData(
                       << "disconnecting_");
     return TransportAdapter::BAD_STATE;
   }
-  sync_primitives::AutoLock locker(out_messages_mutex_);
-  if (current_out_message_.use_count() != 0) {
-    out_messages_.push_back(message);
-  } else {
-    current_out_message_ = message;
-    if (!PostOutTransfer()) {
-      controller_->DataSendFailed(
-          device_uid_, app_handle_, message, DataSendError());
-      LOG4CXX_TRACE(
-          logger_,
-          "exit with TransportAdapter::FAIL. Condition: !PostOutTransfer()");
-      return TransportAdapter::FAIL;
+
+  auto process_message = [&]() {
+    sync_primitives::AutoLock locker(out_messages_mutex_);
+    auto error_code = TransportAdapter::OK;
+    if (current_out_message_.use_count() != 0) {
+      out_messages_.push_back(message);
+    } else {
+      current_out_message_ = message;
+      error_code = PostOutTransfer();
     }
+    return error_code;
+  };
```

Comment:
@SNiukalov I would recommend to make this lambda more short & simple. 
Try to do something like : 
```cpp
auto process_message = [&]() {
  sync_primitives::AutoLock locker(out_messages_mutex_);
  if (current_out_message_.use_count() == 0) {
    current_out_message_ = message;
    return PostOutTransfer();
  } 
    out_messages_.push_back(message);
    return TransportAdapter::OK;
  };
```

Reply:
@AByzhynar 
Thanks, fixed in:https://github.com/smartdevicelink/sdl_core/pull/3332/commits/581a1b9d0e927f22754b6baa59c6df89ec19c721

Reply:
@SNiukalov Check it first - if it works. It was just an example))

Reply:
@AByzhynar 
Thanks, it is working, checked :)